ext/afform/core/Civi/Api4/Action/Afform/Get.php

```diff
@@ -212,6 +218,33 @@ private function getSearchDisplays(string $html) {
     return $searchDisplays;
   }
 
+  /**
+   * Backward-compatability conversion of <af-entity> Contact tags
+   * to use contact-type-specific API entity names.
+   *
+   * Changes this: <af-entity type="Contact" data="{contact_type: 'Individual'}">
+   * Into this:    <af-entity type="Individual" data="{}">
+   *
+   * @param string $layout
+   * @return void
+   */
+  private static function convertLegacyContactTypes(&$layout): void {
+    if (!$layout || !is_string($layout)) {
+      return;
+    }
+    // It was easier to write two regexes:
+    // 1. If the `type` property precedes the `data` property
+    $layout = preg_replace(
+      '/<af-entity([^>]+)type\s*=\s*"Contact"([^>]*)data\s*=\s*"\{([^>]*)contact_type\s*:\s*\'(Individual|Household|Organization)\'\s*,?/',
```

Comment:
I think I found a couple of edge cases:

a) `<af-entity data="{'contact_type':'Individual'}" type="Contact">` doesn't get matched. 

b) `<af-entity data="{contact_type: 'Household'}" my-custom-antidata-attribute="{contact_type:'Individual'}" type="Contact">` gets mapped to an Individual rather than Household.

Quite far-fetched though - and given this is just for backwards compatibility I think it's probably reasonable to expect people to pick up things like this themselves.

Reply:
@ufundo right these regexes aren't perfect but the important thing is the match the format written out by the Afform.save html code generator.

That said, if you have a patch to improve the regexes I'm happy to incorporate it.

Reply:
Yep makes sense 👍